statistics/histogram.go

```diff
@@ -686,6 +690,71 @@ func (hg *Histogram) outOfRange(val types.Datum) bool {
 		chunk.Compare(hg.Bounds.GetRow(hg.Bounds.NumRows()-1), 0, &val) < 0
 }
 
+// MergeHistogramForInnerJoin merges two histogram into one for inner join.
+func MergeHistogramForInnerJoin(lSide *Histogram, rSide *Histogram, tp *types.FieldType) *Histogram {
+	cmpFunc := chunk.GetCompareFunc(tp)
+	lAvgPerVal := lSide.notNullRowCount() / float64(lSide.NDV)
+	rAvgPerVal := rSide.notNullRowCount() / float64(rSide.NDV)
+	totCnt := int64(0)
+	totNdv := float64(0)
+	newHist := NewHistogram(0, 0, 0, 0, tp, 256, 0)
+	for i, j := 0, 0; i < lSide.Bounds.NumRows() && j < rSide.Bounds.NumRows(); {
```

Comment:
how about:
```go
for i, j := 0, 0; i < lSide.Len() && j < rSide.Len(); {
```

use `i` and `j` to specify the id of left hand side bucket and right hand side bucket respectively.

```go
lhsLowerBound, lhsUpperBound := lSide.Bounds.GetRow(2*i), lSide.Bounds.GetRow(2*i+1)
rhsLowerBound, rhsUpperBound := rSide.Bounds.GetRow(2*j), rSide.Bounds.GetRow(2*j+1)
```